core/src/main/java/ca/odell/glazedlists/swing/AutoCompleteSupport.java

```diff
@@ -1520,18 +1705,23 @@ private void selectAutoCompleteTerm(FilterBypass filterBypass, AttributeSet attr
                 }
 
                 // if partial autocompletion terms are not allowed, and we only have a partial term, bail early
-                if (!allowPartialAutoCompletionTerm && !prefix.equals(itemString))
+                if (!allowPartialAutoCompletionTerm && !input.equals(itemString))
                     return;
 
-                // either keep the user's prefix or replace it with the itemString's prefix
+                // either keep the user's input or replace it with the itemString's prefix
                 // depending on whether we correct the case
                 if (getCorrectsCase() || isStrict()) {
-                    filterBypass.replace(0, prefix.length(), matchString, attributeSet);
+                    //filterBypass.replace(0, prefix.length(), matchString, attributeSet);
```

Comment:
I'm not sure with this change. This seems the only place which could break backward compatibility as it applies to strict mode with CONTAINS _and_ STARTS_WITH. I didn't have the time yet to test this extensively.

Reply:
This was one of the first things I did a few weeks ago. Time to revisit.

I think I convinced myself that this didn't really change anything. I was focused on strict and may not have considered getCorrectCase.

Reply:
Yes, your change applies to CorrectsCase, too. But also for strict mode with STARTS_WITH. The latter was my initial concern. Maybe you should make the change only for strict mode with CONTAINS (if required at all)?

Reply:
> Concerning backwards compatibility and the EditorComponent: I just meant that any user can get a reference to the editor:
> JTextField editorComponent = (JTextField) comboBox.getEditor().getEditorComponent()
> This code would break if we'd change to JTextPane in AutoCompleteSupport.

The proposal does not change the EditorComponent. IIUC, glazed does not change the editor component or document, it makes sure they are of the correct type and delegates to them through the `FormatComboBoxEditor`. The proposal is to allow the user to provide an editor that has JTextPane and StyledDocument, glazed will provide additional behavior if this is the case.

There is no backwards compatibility issue. If user code does
```
(JTextField) comboBox.getEditor().getEditorComponent()
``` 
it only fails if the user put in a different kind of text component, something other that a JTextField.

Reply:
At this spot in the code, matchString is the what matches from the combo box list. The code is going to replace the contents of the combo editor, using filterBypass, with the matchString. The old `prefix` variable is the user input *and* what's currently in the combo box. So when the code does
```
filterBypass.replace(0, prefix.length(), matchString, attributeSet)
```
It is replacing the entire contents of the editor.

At least this is my understanding after studying the code and building the table in #696 .  While looking at all the cases, I found out that the original PR was mishandling the non-strict+contains cases.

The code is annotated to clarify the cases.